pkg/api/message/service.go

```diff
@@ -449,7 +453,25 @@ func (s *Service) validateClientInfo(clientEnv *envelopes.ClientEnvelope) error
 		return status.Errorf(codes.InvalidArgument, "topic does not match payload")
 	}
 
-	// TODO(rich): Verify all originators have synced past `last_seen`
+	if aad.GetDependsOn() != nil {
```

Comment:
LGTM.
Should we remove the `aad.GetTargetOriginator() != s.registrant.NodeID()` check on this PR?

Reply:
Yeah. That check will have to go. I need to introduce the new field set by the payer. I suspect we can still validate the originator if it exists for backwards compatibility?

Reply:
Burn the boats. We're too early to worry about backward compatibility for things like validations that don't require coordination between nodes..

Reply:
It's tracked here https://github.com/xmtp/xmtpd/issues/463